### PR TITLE
fix: Global Context

### LINKED ISSUE #928: IOS - Global Context does not provide a friendly API to communicate with it
## Description

Like Android and Web, Beagle for IOS does not provide a friendly api to consume and modify GlobalContext values, like:

This is how it works with Beagle Android:

`GlobalContext.get(path: "myValue")` to get values

`GlobalContext.set(value: "Context has changed", path: "myValue")` to set values

`GlobalContext.clear(path: "myValue")` to clear the value on the path passed by parameter  or if the path is nil clear the global context

## Expected Results
Be able to interact with global context in a simple way like Android and Web

## Code example:
This is what it takes to set a value to a path in the global context:
```
var context = Beagle.dependencies.globalContext.context
var contextValue = context.value.value
contextValue.set("value", forPath: Path(nodes: [.key("path")]))
context.value = Context(id: "global", value: contextValue)
```

### PR DESCRIPTION
**Pull request by heliojuniorzup**
### Related Issues

Closes #928 

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: iOS/Sources/Beagle/Sources/ContextExpression/Tests/GlobalContextTests.swift [hunk 1]
```diff
@@ -27,23 +27,23 @@ final class GlobalContextTests: XCTestCase {
     private var view1 = UIView()
     private var view2 = UIView()
     
-    private lazy var globalContext1 = Context(id: globalId, value: "Fist value")
+    private lazy var globalContext1 = Context(id: globalId, value: "First value")
     private lazy var globalContext2 = Context(id: globalId, value: "Second value")
     
+    private lazy var globalContextWithDictionary = Context(id: globalId, value: .dictionary([globalId : "Dictionary value"]))
```
Comment (botbeaglezup) [inline iOS/Sources/Beagle/Sources/ContextExpression/Tests/GlobalContextTests.swift:33]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/b7a52e1adef9003bc1e0600c761403765bfdef66/iOS/Sources/Beagle/Sources/ContextExpression/Tests/GlobalContextTests.swift#L33"></span>Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.<br />
<code>colon</code> <code>GlobalContextTests.swift:33</code></td>
    </tr>
  </tbody>
</table>

**Review by caioortu — DISMISSED** [review]

File: iOS/Sources/Beagle/Sources/ContextExpression/GlobalContext.swift [hunk 2]
```diff
@@ -41,7 +43,23 @@ public class DefaultGlobalContext: GlobalContext {
...
+        var contextValue = context.value.value
+        contextValue.set(value, forPath: pathObject)
+        context.value = Context(id: globalId, value: contextValue)
+    }
+    
+    public func get(path: String? = nil) -> DynamicObject {
+        let pathObject = Path(rawValue: path ?? "")
+//        return pathObject.evaluate
+        return nil
+    }
```
Comment (yandiaszup) [inline iOS/Sources/Beagle/Sources/ContextExpression/GlobalContext.swift:60]: There is a comment here and this function its aways returning nil

**Review by yandiaszup — CHANGES_REQUESTED** [review]

File: iOS/Sources/Beagle/Sources/ContextExpression/Tests/GlobalContextTests.swift [hunk 3]
```diff
@@ -27,23 +27,22 @@ final class GlobalContextTests: XCTestCase {
     private var view1 = UIView()
     private var view2 = UIView()
     
-    private lazy var globalContext1 = Context(id: globalId, value: "Fist value")
+    private lazy var globalContext1 = Context(id: globalId, value: "First value")
     private lazy var globalContext2 = Context(id: globalId, value: "Second value")
     
+    private lazy var globalContextWithDictionary = Context(id: globalId, value: .dictionary([globalId: "First value"]))
+    
     func testGetContext() {
```
Comment (lucasfonsecazup) [inline iOS/Sources/Beagle/Sources/ContextExpression/Tests/GlobalContextTests.swift:35]: We are adopting the comments `// Given // When // Then` as convention -> https://github.com/ZupIT/beagle/wiki/Swift-Style-Guide#tests

**Review by lucasfonsecazup — COMMENTED** [review]
The solution is very good, however there are some points.

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by caioortu — APPROVED** [review]

**Review by yandiaszup — APPROVED** [review]

**Review by gabcoelho — APPROVED** [review]

**Review by lucasaraujo — APPROVED** [review]

**Pull request merged**